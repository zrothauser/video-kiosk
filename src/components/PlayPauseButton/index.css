.b-play-pause-button {
  -webkit-appearance: none;
  background: transparent;
  border: 0;
  width: 7.8540104167vw;
  height: 7.8540104167vw;
  position: absolute;
  z-index: 10;
  top: calc(50% - 3.9270052084vw);
  left: calc(50% - 3.9270052084vw);
  cursor: pointer;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  -webkit-tap-highlight-color: transparent;
}

.b-play-pause-button:focus,
.b-play-pause-button:active {
  outline: 0;
}

.b-play-pause-button img:focus {
  outline: 0;
}
