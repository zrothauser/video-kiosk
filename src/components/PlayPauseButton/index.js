// Dependencies
import React from 'react';
import PropTypes from 'prop-types';
import Transition from 'react-transition-group/Transition';

// Styles/Resources
import './index.css';
import PlayButtonSVG from '../../resources/icons/play.svg';
import PauseButtonSVG from '../../resources/icons/pause.svg';

// Transition
import * as transitions from '../transitions';

const {
  durations,
  layerAStyles,
} = transitions;

class PlayPauseButton extends React.Component {
  static renderPlayButton() {
    return (
      <img src={PlayButtonSVG} alt="Play" />
    );
  }

  static renderPauseButton() {
    return (
      <img src={PauseButtonSVG} alt="Pause" />
    );
  }

  render() {
    const {
      isPlaying,
      togglePlay,
      visible,
    } = this.props;

    // Build content
    let buttonContent;

    if (isPlaying) {
      buttonContent = PlayPauseButton.renderPauseButton();
    } else {
      buttonContent = PlayPauseButton.renderPlayButton();
    }

    return (
      <Transition
        in={visible}
        timeout={durations.medium}
      >
        {state => (
          <button
            className="b-play-pause-button"
            onClick={() => togglePlay(!isPlaying)}
            style={{
              ...layerAStyles.default,
              ...layerAStyles[state],
            }}
          >
            {buttonContent}
          </button>
        )}
      </Transition>
    );
  }
}

PlayPauseButton.propTypes = {
  isPlaying: PropTypes.bool.isRequired,
  togglePlay: PropTypes.func.isRequired,
  visible: PropTypes.bool.isRequired,
};

export default PlayPauseButton;
