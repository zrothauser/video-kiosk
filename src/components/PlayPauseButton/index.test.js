import React from 'react';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import PlayPauseButton from './index';

describe('components', () => {
  describe('PlayPauseButton', () => {
    const mockProps = {
      isPlaying: true,
      togglePlay: jest.fn(),
      visible: true,
    };

    it('renders without crashing', () => {
      shallow(<PlayPauseButton {...mockProps} />);
    });

    it('renders correctly', () => {
      const PlayPauseButtonComponent = renderer.create(
        <PlayPauseButton {...mockProps} />,
      );

      expect(PlayPauseButtonComponent).toMatchSnapshot();
    });
  });
});
