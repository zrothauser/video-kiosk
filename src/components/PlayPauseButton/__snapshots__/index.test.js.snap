// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components PlayPauseButton renders correctly 1`] = `
<button
  className="b-play-pause-button"
  onClick={[Function]}
  style={
    Object {
      "opacity": 0,
      "transition": "opacity 400ms ease-in-out",
    }
  }
>
  <img
    alt="Pause"
    src="pause.svg"
  />
</button>
`;
